### PR TITLE
Show filter segment used in listings/stats [MAILPOET-5511]

### PR DESCRIPTION
**Pull request by johnolek**
## Description

This PR adds a tag on listing pages/stats pages for newsletters that support filter segments if the email is currently configured to use a filter segment or if the email was sent using a filter segment.

Currently the UI for adding a filter segment to an email is not merged, but you can use the testing directions from https://github.com/mailpoet/mailpoet/pull/5108 to test this immediately.

The tooltip is going to be updated to include specific conditions in a separate ticket.

## Code review notes

I realized during my own testing that the filter segment data wasn't getting saved to the queue for post notification emails, and this PR fixes that issue by having the filter segment getter check the parent newsletter. This seemed cleaner to me than creating a new option in the database when creating the child newsletter.

## QA notes

For sent emails, the filter segment is stored on the sending queue itself, including the name. This means that if the segment name is changed we will continue to see the old segment name for sent emails. This also ensures we can continue to display the filter segment name if the segment gets deleted. Please make sure to test these cases, though it will change in the future to include more data.

I wasn't sure of a good way to test re-engagement emails locally, so please make sure to test that they get tested.

While working on this I noticed a bug where having premium installed but not active makes it impossible to view a newsletter's stats page. The plugin still tries to hit a premium-only endpoint, resulting in an `Invalid API Endpoint` error. This PR includes a fix for this and I would appreciate if you could test this as well.

image

## Linked PRs

You can use #5175 to add filter segments to emails.

## Linked tickets

https://mailpoet.atlassian.net/browse/MAILPOET-5512

## After-merge notes

_N/A_

## Tasks

- [x] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] I added sufficient test coverage
- [x] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

### CODE REVIEW

**Comment by johnolek** [comment]
@rodrigoprimo I believe the tests are only failing because of the current issue with dockerhub. I'll try re-running them again tomorrow if the one running now doesn't finish. I don't expect there to be any true failures but feel free to wait until all is green before starting to review.

**Comment by rodrigoprimo** [comment]
@johnolek, just FYI, this PR needs to be rebased and some conflicts fixed after https://github.com/mailpoet/mailpoet/pull/5184 was merged.

File: mailpoet/lib/Entities/NewsletterEntity.php
Comment (rodrigoprimo) [inline mailpoet/lib/Entities/NewsletterEntity.php:502]: Would it make sense to check the newsletter type to make sure this is only applied for post notifications? Maybe it doesn't. I'm not sure.

Also, I'm curious as to why we need to check if `$this->getId() !== $parentNewsletter->getId()`? Is there a scenario where `$this->getId() === $parentNewsletter->getId()`?

File: mailpoet/lib/AdminPages/Pages/Newsletters.php
Comment (rodrigoprimo) [inline mailpoet/lib/AdminPages/Pages/Newsletters.php:130]: Good catch! Would it make sense to highlight this commit in the QA notes so that QA is aware of this fix and can test it if they consider necessary?

**Review by rodrigoprimo — APPROVED** [review]
Looks good to me! I just left two small comments, but they are not blockers. Feel free to move this directly to QA if you prefer.

File: mailpoet/lib/AdminPages/Pages/Newsletters.php
Comment (johnolek) [reply mailpoet/lib/AdminPages/Pages/Newsletters.php:130]: Good call! I'll update the description.

File: mailpoet/lib/Entities/NewsletterEntity.php
Comment (johnolek) [reply mailpoet/lib/Entities/NewsletterEntity.php:502]: > Would it make sense to check the newsletter type to make sure this is only applied for post notifications?

My inclination is to keep this as is. I'm pretty sure parent newsletters are only used for post notifications, but if we had others or add others in the future it seems like it would make sense to defer to the parent's filter segment if the child doesn't have one of its own. I don't think it would make much of a difference right now though.

> Also, I'm curious as to why we need to check if $this->getId() !== $parentNewsletter->getId()? Is there a scenario where $this->getId() === $parentNewsletter->getId()?

Thanks for calling this out. There are a couple of tests that set the parent newsletter to itself ([example1](https://github.com/mailpoet/mailpoet/blob/360d9410f915b97c3adb053948167e2552b55fbc/mailpoet/tests/integration/Cron/Workers/SendingQueue/Tasks/NewsletterTest.php#L221-L221), [example2](https://github.com/mailpoet/mailpoet/blob/360d9410f915b97c3adb053948167e2552b55fbc/mailpoet/tests/integration/Cron/Workers/SendingQueue/Tasks/NewsletterTest.php#L240-L240)) that also preprocess the newsletter, resulting in an infinite loop. I'm almost positive we would never see this in the wild, but I would rather play it on the safe side.

**Review by Aschepikov — APPROVED** [review]
GJ!

**Pull request merged**
